testsuite/features/step_definitions/api_common.rb

```diff
@@ -178,6 +178,36 @@
   assert_equal(false, ret.key?(attr))
 end
 
+Then(/^channel "([^"]*)" should (not )?be enabled on "([^"]*)"$/) do |channel, disabled, host|
+  node = get_target(host)
+  system_id = get_system_id(node)
+
+  channels = $api_test.channel.software.list_system_channels(system_id)
+  assert_equal(false, channels.empty?)
+
+  assert_equal(disabled.nil?, channels.include?(channel))
+end
+
+Then(/^"(\d+)" channels should be enabled on "([^"]*)"$/) do |count, host|
+  node = get_target(host)
+  system_id = get_system_id(node)
```

Comment:
same, it sounds to me like we're mixing ssh and api

Reply:
Sure we are mixing.
From one side, we want to obtain the twopence node, and from the other side we need the system id of that node.
We could be storing the id inside the Lavanda class, but that would be another refactor, and we need to be sure to keep this id value up to date in all moment (when we delete and recreate systems)

Reply:
Looking also at the other comment about API calls, I think we're fine with the current implementation and can refactor this part later on if needed.
This PR is really more about handling the checks on channels .